gdpr_assist/models.py

```diff
@@ -246,33 +278,278 @@ def _cast_class(cls, model, privacy_meta):
 
         return model
 
+    def get_anon_log(self):
+        """
+        See if there are any related models which need to be anonymised.
+
+        They will be any reverse relations to PrivacyModel subclasses where their
+        OneToOneField and ForeignKey on_delete is ANONYMISE.
+        """
+        if not self.anonymised:
+            return "%s is not anonymised yet." % self
+
+        from gdpr_assist.models import EventLog
+        logs = EventLog.objects.order_by('log_time')
+
+        top_level_log_lines = EventLog.objects.for_instance(self).values('log_time', 'event', 'app_label', 'model_name',
+                                                                        'target_pk', 'acting_user')
+
+        if top_level_log_lines.count() > 0:
+            actual_anon_log_line_start = top_level_log_lines.filter(event=EventLog.EVENT_RECURSIVE_START).last()
+            actual_anon_log_line_end = top_level_log_lines.filter(event=EventLog.EVENT_ANONYMISE).last()
+
+            lines = logs.filter(log_time__gte=actual_anon_log_line_start["log_time"],
+                                log_time__lte=actual_anon_log_line_end["log_time"]). \
+                values('log_time', 'event', 'app_label', 'model_name', 'target_pk', 'acting_user')
+
+            user = str(actual_anon_log_line_start["acting_user"]) if actual_anon_log_line_start[
+                                                                   "acting_user"] is not None else "[Non-descript user]"
+
+            res = "%s #%s starting to anonymise [by %s on %s].\n" % (
+                actual_anon_log_line_start['model_name'],
+                actual_anon_log_line_start['target_pk'],
+                user,
+                actual_anon_log_line_start["log_time"].strftime("%Y-%m-%d %H:%M:%S")
+            )
+            indent_level = 0
+
+            def tabify(x):
+                return "\t" * x if x > 0 else ""
+
+            for l in lines:
+                if l["event"] == EventLog.EVENT_RECURSIVE_START:
+                    res += "%sStarting recursive for %s #%s.\n" % (
+                    tabify(indent_level), l['model_name'], l['target_pk'])
+                    indent_level += 1
+
+                elif l["event"] == EventLog.EVENT_RECURSIVE_END:
+                    indent_level -= 1
+                    res += "%sEnding recursive for %s #%s.\n" % (tabify(indent_level), l['model_name'], l['target_pk'])
+
+                elif l["event"] == EventLog.EVENT_ALREADY_ANONYMISED:
+                    res += "%s%s #%s already anonymised.\n" % (tabify(indent_level), l['model_name'], l['target_pk'])
+
+                else:
+                    res += "%s%s #%s flat fields anonymised.\n" % (
+                    tabify(indent_level), l['model_name'], l['target_pk'])
+
+            return res
+        else:
+            return "%s is anonymised, but does not have matching logs." % self
+
+    @classmethod
+    def get_anonymization_tree(cls, prefix="", doprint=False, objs=[]):
```

Comment:
European style for name?
Also, objs doesn't seem to be used here? Is it a legacy thing where subclasses could use it?
It makes me cringe when I see a mutable object as default value. Can we default it to `None` instead and do the `if objs is None: objs = []` dance instead? 

```suggestion
    def get_anonymisation_tree(cls, prefix="", doprint=False, objs=[]):
```

Reply:
Why? An empty list is a totally reasonable default. And re the `objs` argument, it is used b the `RetentionPolicy` override of this method (and theoretically could be used to built object-specific lists of "what am i about to anonymise (recursively)